src/TriggerBinding/SqlScalerProvider.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql
+{
+    internal class SqlScalerProvider : IScaleMonitorProvider, ITargetScalerProvider
+    {
+        private readonly SqlTriggerScaleMonitor _scaleMonitor;
+        private readonly SqlTriggerTargetScaler _targetScaler;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SqlScalerProvider"/> class.
+        /// Use for obtaining scale metrics by scale controller.
+        /// </summary>
+        /// <param name="serviceProvider"></param>
+        /// <param name="triggerMetadata"></param>
+        public SqlScalerProvider(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata)
+        {
+            IConfiguration config = serviceProvider.GetService<IConfiguration>();
+            ILoggerFactory loggerFactory = serviceProvider.GetService<ILoggerFactory>();
+            ILogger logger = loggerFactory.CreateLogger(LogCategories.CreateTriggerCategory("Sql"));
+            SqlMetaData sqlMetadata = JsonConvert.DeserializeObject<SqlMetaData>(triggerMetadata.Metadata.ToString());
+            sqlMetadata.ResolveProperties(serviceProvider.GetService<INameResolver>());
+            var userTable = new SqlObject(sqlMetadata.TableName);
+            string connectionString = SqlBindingUtilities.GetConnectionString(sqlMetadata.ConnectionStringSetting, config);
+            int maxChangesPerWorker = config.GetValue(SqlTriggerConstants.ConfigKey_SqlTrigger_MaxChangesPerWorker, SqlOptions.DefaultMaxChangesPerWorker);
+            string userDefinedLeasesTableName = sqlMetadata.LeasesTableName;
+            string userFunctionId = (string)triggerMetadata.Properties["userFunctionId"];
```

Comment:
Are we guaranteed that this is going to have a value? Is there a reason you're not able to use the same method we use for getting the function ID elsewhere?

https://github.com/Azure/azure-functions-sql-extension/blob/main/src/TriggerBinding/SqlTriggerBinding.cs#L105

Reply:
Why don't we use `triggerMetadata.FunctionName` here?

Reply:
@Charles-Gagnon we can't really reuse the function ID from the SqlTriggerBinding, can I replace it with  `triggerMetadata.FunctionName` mentioned above?

Reply:
Why aren't we able to use the same method? Do we not have access to IHostIdProvider ?

If not, then I believe we're still going to need to find another since we still will need to get the function ID somehow in order for the metrics provider to be able to get the list of unprocessed changes : https://github.com/Azure/azure-functions-sql-extension/blob/main/src/TriggerBinding/SqlTriggerMetricsProvider.cs#L103

Reply:
the userFunctionId is gotten from HostIdProvider, and it would use this to  
https://github.com/Azure/azure-functions-host/blob/47411dfdc56bedd7f7dd20aa487fe69dbc596325/src/WebJobs.Script/Host/ScriptHostIdProvider.cs#L49

I saw that the scalecontroller also had a hostid which I thought I could use, but the values might be different, as it is using a different class namely FixedHostIdProvider please confirm @alrod 

We could go ahead and use triggerMetadata.FunctionName but we would have to use the same at places where userFunctionId is used, in order to get unprocessed changes, as Charles has mentioned.